app/Svg/Parser.hs

```diff
@@ -286,6 +289,39 @@ readAttr attr tag = fromMaybe
     (error ("reading " ++ T.unpack attr ++ " from " ++ show tag))
     (readMaybe $ T.unpack $ fromAttrib attr tag)
 
+-- | Runs a parser on a text object.
+-- Throws an exception on any parse errors.
+parseVal :: Parser a -> T.Text -> a
+parseVal parser input =
+    case P.parse parser "" strippedInput of
+        Left err ->
+            error ("reading " ++ T.unpack input ++ ":" ++ show err)
+        Right val -> val
+    where strippedInput = filter (not . isSpace) (T.unpack input)
```

Comment:
Don't strip out whitespace here. Instead, modify your parsers to allow optional whitespace (`Text.Parsec.Char` has a `spaces` parser) between the key elements.